R/package.R

```diff
@@ -32,16 +33,22 @@ is_python_initialized <- function() {
 
 
 ensure_python_initialized <- function(required_module = NULL) {
+
   if (!is_python_initialized()) {
-     # give delay load modules priority
-     use_environment <- NULL
-     if (!is.null(.globals$delay_load_module)) {
-        required_module <- .globals$delay_load_module
-        use_environment <- .globals$delay_load_environment
-        .globals$delay_load_module <- NULL # one shot
-        .globals$delay_load_environment <- NULL
-        .globals$delay_load_priority <- 0
-     }
+    # give delay load modules priority
+    use_environment <- NULL
+    if (!is.null(.globals$delay_load_module)) {
+      required_module <- .globals$delay_load_module
+      use_environment <- .globals$delay_load_environment
+      .globals$delay_load_module <- NULL # one shot
+      .globals$delay_load_environment <- NULL
+      .globals$delay_load_priority <- 0
+    }
+
+    # mark as initializing (so we know to save any PATH changes)
+    .globals$py_initializing <- TRUE
```

Comment:
`py_disover_config()` can potentially call `py_config()` multiple times, which would in turn cause the `PATH` to be amended multiple times during initialization. I think that rather than using the `py_initializing` flag we should package the PATH manipulating code into a helper function and call it from both places.